File: Python/BAEKJOON/Silver Ⅳ/2485 가로수.py
import sys
input = sys.stdin.readline

# 유클리드 호제법을 사용한 최대공약수 계산
def GCD(x, y):
    while x % y != 0:
        x, y = y, x % y

    return y

N = int(input())

positions = [int(input()) for _ in range(N)]
size = N - 1

# 연속된 가로수들 사이의 거리 계산
gaps = [abs(positions[i + 1] - positions[i]) for i in range(size)]

# 모든 간격의 최대공약수 계산
gcd = gaps[0]
total_distance = gaps[0]

for i in range(1, size):
    gcd = GCD(gcd, gaps[i])
    total_distance += gaps[i]

# 필요한 가로수 개수 = (총 거리 / 최소 간격) - 기존 간격 개수
print(total_distance // gcd - size)

File: javascript/BAEKJOON/SILVERⅣ/2485 가로수.js
var input = require('fs').readFileSync(__dirname+'/test.txt').toString().trim().split("\n");
//var input = require('fs').readFileSync("/dev/stdin", "utf-8").trim().split("\n");

/**
 * 유클리드 호제법을 사용한 최대공약수 계산
 */
const GCD = (x, y) => {
    let r;
    while(x % y != 0){

        r = x % y;
        x = y;
        y = r;
    }

    return y;
}


const N = parseInt(input[0]);

let positions = new Array(N);

for (let i = 1; i <= N; i++) {
    positions[i - 1] = parseInt(input[i]);
}

// 연속된 가로수들 사이의 거리 계산
let gaps = new Array(N - 1);
for(let i = 0; i < N - 1; i++) {
    gaps[i] = Math.abs(positions[i + 1] - positions[i]);
}

// 모든 간격의 최대공약수 계산
let gcd = gaps[0];
let totalDistance = gaps[0];
for (let i = 1; i < gaps.length; i++) {
    gcd = GCD(gcd, gaps[i]);
    totalDistance += gaps[i];
}

// 필요한 가로수 개수 = (총 거리 / 최소 간격) - 기존 간격 개수
console.log((totalDistance / gcd) - gaps.length);